### PR TITLE
XW-2384 | exchange of image preview and add image actions in adding media modal

### PR DESCRIPTION
**Pull request by weronika**
https://wikia-inc.atlassian.net/browse/XW-2384

### CODE REVIEW

File: extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaVideoOptionWidget.js [hunk 1]
```diff
@@ -14,20 +14,20 @@
  * @cfg {number} [size] Media thumbnail size
  */
 ve.ui.WikiaVideoOptionWidget = function VeUiWikiaVideoOptionWidget( config ) {
```
Comment (WikiaTech) [inline extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaVideoOptionWidget.js]: MAJOR Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* CRITICAL 1 critical
* MAJOR 8 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [ve.ui.WikiaMediaInsertDialog.js#L660](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L660): TypeError can be thrown as "count" might be null or undefined here. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2259)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L14](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L14): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L259](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L259): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L621](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L621): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L646](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L646): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.ui.WikiaMediaInsertDialog.js#L649](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L649): Remove this assignment of "type". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2310)
1. MAJOR [ve.ui.WikiaMediaOptionWidget.js#L19](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaMediaOptionWidget.js#L19): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.ui.WikiaMediaOptionWidget.js#L104](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaMediaOptionWidget.js#L104): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. INFO [ve.ui.WikiaMediaInsertDialog.js#L529](https://github.com/Wikia/app/blob/14c58d84d584191b5a3425e2e45ef6513ee50676/extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js#L529): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

File: extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaMediaOptionWidget.js [hunk 2]
```diff
@@ -34,17 +34,17 @@ ve.ui.WikiaMediaOptionWidget = function VeUiWikiaMediaOptionWidget( config ) {
...
 		frameless: true
 	} );
 
 	// Events
 	this.$image
 		.load( this.onThumbnailLoad.bind( this ) )
 		.error(this.onThumbnailError.bind( this ) );
-	this.check.on( 'click', function () {
+	this.previewButton.on( 'click', function () {
 		this.emit( 'check', this );
```
Comment (rogatty) [inline extensions/VisualEditor/wikia/modules/ve/ui/widgets/ve.ui.WikiaMediaOptionWidget.js:48]: Can we rename this too? IMO it should emit `preview` event.

File: extensions/VisualEditor/wikia/modules/ve/ui/styles/images/icons/preview-photo.svg [hunk 3]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="iso-8859-1"?>
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
 <svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
-	 width="14px" height="14px" viewBox="0 0 14 14" enable-background="new 0 0 14 14" xml:space="preserve">
-<g fill="#7f7f7f">
+	 width="24px" height="24px" viewBox="0 0 14 14" enable-background="new 0 0 14 14" xml:space="preserve">
```
Comment (rogatty) [inline extensions/VisualEditor/wikia/modules/ve/ui/styles/images/icons/preview-photo.svg:4]: Are these modified svgs used anywhere else? If yes, we should check how they look now.

File: extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js [hunk 4]
```diff
@@ -140,8 +140,8 @@ ve.ui.WikiaMediaInsertDialog.prototype.initialize = function () {
 	this.queryInput.$input.on( 'keydown', this.onQueryInputKeydown.bind( this ) );
 	this.search.connect( this, {
 		nearingEnd: 'onSearchNearingEnd',
-		check: 'onSearchCheck',
-		select: 'onMediaPreview'
+		select: 'onSearchCheck',
+		check: 'onMediaPreview'
```
Comment (rogatty) [inline extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaMediaInsertDialog.js:144]: This looks confusing, let's unify the naming.

**Review by rogatty — CHANGES_REQUESTED** [review]

File: extensions/VisualEditor/wikia/modules/ve/ui/styles/images/icons/preview-photo.svg (hunk 3 above)
Comment (weronika) [reply extensions/VisualEditor/wikia/modules/ve/ui/styles/images/icons/preview-photo.svg:4]: I've already checked that - we don't use them in any other place

**Pull request closed without merging**